jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/validation/BPMN2SyntaxChecker.java

```diff
@@ -592,6 +594,12 @@ private void checkDataObject(DataObject dataObject) {
 		}
 	}
 
+	private void checkMandatoryDefaultGate(Gateway gateway, SequenceFlow defaultSequenceFlow) {
+		if(defaultSequenceFlow == null || (gateway.getOutgoing() != null && !gateway.getOutgoing().contains(defaultSequenceFlow))) {
+			addError(gateway, new ValidationSyntaxError(gateway, BPMN2_TYPE, SyntaxCheckerErrors.NOT_VALID_DEFAULT_GATE));
+		}
+	}
+
```

Comment:
Default Gate should always be optional. There is no hard requirement to have it set in the runtime as well as during process modeling. It is OK if the default gate is not specified so this check method is not valid.

What you can check is if a gateway does have a default specified if indeed it is one of the outgoing flows of it. That would be OK.